src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```diff
@@ -136,7 +138,10 @@ public boolean isBootstrapRequired(Map<String, Object> newConfig)
             }
         });
         if (componentsRequiresBootstrapTask.isEmpty()) {
-            return nucleusConfigValidAndNeedsRestart;
+            // Force restart if
+            // 1. any nucleus config change requires restart or
+            // 2. if any plugin will be removed in the deployment to ensure plugin cleanup
+            return nucleusConfigValidAndNeedsRestart || willRemovePlugins(serviceConfig);
```

Comment:
so this will return that bootstrap is needed, but will it actually cause it to restart? Don't we need to tell it somewhere that the bootstrap that we want is a restart?

Reply:
This definitely could be right, I'm just not confident without real testing.

Reply:
It'll by default be a restart, we already do this for things like when runWith config changes, which is a similar in that it is not a real bootstrap task as such